packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/service/ExpoNotificationsService.java

```diff
@@ -198,10 +220,38 @@ protected Notification getNotification(StatusBarNotification statusBarNotificati
         String message = String.format("Could not have unmarshalled NotificationRequest from (%s, %d).", statusBarNotification.getTag(), statusBarNotification.getId());
         Log.e("expo-notifications", message);
       }
+    } else {
+      // It's not our notification. Let's do what we can.
+      NotificationContent content = new NotificationContent.Builder()
+        .setTitle(notification.extras.getString(android.app.Notification.EXTRA_TITLE))
+        .setText(notification.extras.getString(android.app.Notification.EXTRA_TEXT))
+        .setSubtitle(notification.extras.getString(android.app.Notification.EXTRA_SUB_TEXT))
+        // using deprecated field
+        .setPriority(NotificationPriority.fromNativeValue(notification.priority))
+        // using deprecated field
+        .setVibrationPattern(notification.vibrate)
+        // using deprecated field
+        .setSound(notification.sound)
+        .setAutoDismiss((notification.flags & android.app.Notification.FLAG_AUTO_CANCEL) != 0)
+        .setBody(fromBundle(notification.extras))
+        .build();
+      NotificationRequest request = new NotificationRequest(getInternalIdentifierKey(statusBarNotification), content, null);
+      return new Notification(request, new Date(statusBarNotification.getPostTime()));
     }
     return null;
   }
 
+  protected JSONObject fromBundle(Bundle bundle) {
+    JSONObject json = new JSONObject();
+    for (String key : bundle.keySet()) {
+      try {
+        json.put(key, JSONObject.wrap(bundle.get(key)));
+      } catch (JSONException e) {
+        // can't do anything about it
```

Comment:
So maybe we could log the exception?

Reply:
Fixed at https://github.com/expo/expo/pull/8614/commits/50724c2976308aa37b6fa1073811f8c9c1969c89!